homeassistant/components/vicare/number.py

```diff
@@ -0,0 +1,217 @@
+        min_value_getter=lambda api: api.getProgramMinTemperature("comfort"),
+        max_value_getter=lambda api: api.getProgramMaxTemperature("comfort"),
+        stepping_getter=lambda api: api.getProgramStepping("comfort"),
+    ),
+)
+
+
+def _build_entity(
+    name: str,
+    vicare_api: PyViCareHeatingDeviceWithComponent,
+    device_config: PyViCareDeviceConfig,
+    entity_description: ViCareNumberEntityDescription,
+) -> ViCareNumber | None:
+    """Create a ViCare number entity."""
+    if is_supported(name, entity_description, vicare_api):
+        _LOGGER.debug("Found device %s", name)
+        return ViCareNumber(
+            name,
+            vicare_api,
+            device_config,
+            entity_description,
+        )
+    return None
+
+
+async def _entities_from_descriptions(
+    hass: HomeAssistant,
+    entities: list[ViCareNumber],
+    iterables: list[PyViCareHeatingDeviceWithComponent],
+    config_entry: ConfigEntry,
+) -> None:
+    """Create entities from descriptions and list of circuits."""
+    for current in iterables:
+        suffix = ""
+        if len(iterables) > 1:
+            suffix = f" {current.id}"
+
+        for description in PROGRAM_ENTITY_DESCRIPTIONS:
+            entity = await hass.async_add_executor_job(
+                _build_entity,
+                f"{description.name}{suffix}",
+                current,
+                hass.data[DOMAIN][config_entry.entry_id][VICARE_DEVICE_CONFIG],
+                description,
+            )
+            if entity is not None:
+                entities.append(entity)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create the ViCare number devices."""
+    api = hass.data[DOMAIN][config_entry.entry_id][VICARE_API]
+
+    entities: list[ViCareNumber] = []
+    try:
+        await _entities_from_descriptions(hass, entities, api.circuits, config_entry)
+    except PyViCareNotSupportedFeatureError:
+        _LOGGER.debug("No circuits found")
+
+    async_add_entities(entities)
+
+
+class ViCareNumber(ViCareEntity, NumberEntity):
+    """Representation of a ViCare number."""
+
+    entity_description: ViCareNumberEntityDescription
+
+    def __init__(
+        self,
+        name: str,
+        api: PyViCareHeatingDeviceWithComponent,
+        device_config: PyViCareDeviceConfig,
+        description: ViCareNumberEntityDescription,
+    ) -> None:
+        """Initialize the number."""
+        super().__init__(device_config)
+        self.entity_description = description
+        self._attr_name = name
+        self._api = api
+        self._device_config = device_config
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._attr_native_value is not None
+
+    @property
+    def unique_id(self) -> str:
```

Comment:
We can just set this once during `__init__`

Reply:
addressed with https://github.com/home-assistant/core/pull/104277